### PR TITLE
Fixing-customize

### PR DESCRIPTION
**Pull request by ofiriro3**
- Fixes: https://github.com/elastic/cloudbeat/issues/332

### CODE REVIEW

File: deploy/k8s/kustomize/base/cloudbeat-ds-eks.yml [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+  namespace: kube-system
+  labels:
+    k8s-app: cloudbeat
+data:
+  cloudbeat.yml: |-
+    cloudbeat:
+      # Defines how often an event is sent to the output
+      type: cloudbeat/eks
+      streams:
+        - access_key_id: <your-aws-access-key>
```
Comment (ofiriro3) [inline deploy/k8s/kustomize/base/cloudbeat-ds-eks.yml:138]: Is it clear enough to a developer that he will need to replace these values?

File: deploy/kustomize/base/daemonset.yml [hunk 2]
```diff
@@ -21,20 +21,10 @@ spec:
           image: cloudbeat:latest
           imagePullPolicy: IfNotPresent
           env:
-            - name: AWS_ACCESS_KEY
```
Comment (ofiriro3) [inline deploy/kustomize/base/daemonset.yml:24]: I added the environment variables in the EKS-deamonset overlay file

**Review by ofiriro3 — COMMENTED** [review]
Good luck!

**Comment by DaveSys911** [comment]
I tried deploy the yaml on eks. Rendered it & added my aws creds. cloudbeat crashes due to some panic error i didn't see before.

`{"log.level":"fatal","@timestamp":"2022-08-16T01:38:10.079Z","log.logger":"cloudbeat","log.origin":{"file.name":"instance/beat.go","file.line":171},"message":"Failed due to panic.","service.name":"cloudbeat","error":{"message":"runtime error: index out of range [0] with length 0"},"stack":"github.com/elastic/beats/v7/libbeat/cmd/instance.Run.func1.1\n\t/Users/daveops/.gvm/pkgsets/go1.18.2/global/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20220726163833-e8a0da132f1f/libbeat/cmd/instance/beat.go:172\nruntime.gopanic\n\t/Users/daveops/.gvm/gos/go1.18.2/src/runtime/panic.go:838\nruntime.goPanicIndex\n\t/Users/daveops/.gvm/gos/go1.18.2/src/runtime/panic.go:89\ngithub.com/elastic/cloudbeat/resources/fetchersManager.addCredentialsToFetcherConfiguration\n\t/Users/daveops/git_repos/elastic/cloudbeat/resources/fetchersManager/factory.go:109\ngithub.com/elastic/cloudbeat/resources/fetchersManager.(*factories).ParseConfigFetchers\n\t/Users/daveops/git_repos/elastic/cloudbeat/resources/fetchersManager/factory.go:67\ngithub.com/elastic/cloudbeat/beater.initRegistry\n\t/Users/daveops/git_repos/elastic/cloudbeat/beater/cloudbeat.go:217\ngithub.com/elastic/cloudbeat/beater.NewCloudbeat\n\t/Users/daveops/git_repos/elastic/cloudbeat/beater/cloudbeat.go:99\ngithub.com/elastic/cloudbeat/launcher.(*launcher).runBeater\n\t/Users/daveops/git_repos/elastic/cloudbeat/launcher/launcher.go:127\ngithub.com/elastic/cloudbeat/launcher.(*launcher).run\n\t/Users/daveops/git_repos/elastic/cloudbeat/launcher/launcher.go:105\ngithub.com/elastic/cloudbeat/launcher.(*launcher).Run\n\t/Users/daveops/git_repos/elastic/cloudbeat/launcher/launcher.go:99\ngithub.com/elastic/beats/v7/libbeat/cmd/instance.(*Beat).launch\n\t/Users/daveops/.gvm/pkgsets/go1.18.2/global/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20220726163833-e8a0da132f1f/libbeat/cmd/instance/beat.go:475\ngithub.com/elastic/beats/v7/libbeat/cmd/instance.Run.func1\n\t/Users/daveops/.gvm/pkgsets/go1.18.2/global/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20220726163833-e8a0da132f1f/libbeat/cmd/instance/beat.go:180\ngithub.com/elastic/beats/v7/libbeat/cmd/instance.Run\n\t/Users/daveops/.gvm/pkgsets/go1.18.2/global/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20220726163833-e8a0da132f1f/libbeat/cmd/instance/beat.go:181\ngithub.com/elastic/beats/v7/libbeat/cmd.genRunCmd.func1\n\t/Users/daveops/.gvm/pkgsets/go1.18.2/global/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20220726163833-e8a0da132f1f/libbeat/cmd/run.go:36\ngithub.com/spf13/cobra.(*Command).execute\n\t/Users/daveops/.gvm/pkgsets/go1.18.2/global/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:876\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/Users/daveops/.gvm/pkgsets/go1.18.2/global/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990\ngithub.com/spf13/cobra.(*Command).Execute\n\t/Users/daveops/.gvm/pkgsets/go1.18.2/global/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918\nmain.main\n\t/Users/daveops/git_repos/elastic/cloudbeat/main.go:29\nruntime.main\n\t/Users/daveops/.gvm/gos/go1.18.2/src/runtime/proc.go:250","ecs.version":"1.6.0"}`

**Review by DaveSys911 — COMMENTED** [review]
See comment about errors while testing

File: deploy/k8s/kustomize/base/cloudbeat-ds-eks.yml (hunk 1 above)
Comment (ofiriro3) [reply deploy/k8s/kustomize/base/cloudbeat-ds-eks.yml]: > I tried deploy the yaml on eks by checking out the pr.Rendered it & added my aws creds. cloudbeat crashes due to some panic error i didn't see before.How did u test this?
> 
> `"log.level":"fatal","@timestamp":"2022-08-16T01:43:00.472Z","log.logger":"cloudbeat","log.origin":{"file.name":"instance/beat.go","file.line":171},"message":"Failed due to panic.","service.name":"cloudbeat","error":{"message":"runtime error: index out of range [0] with length 0"},"stack":"github.com/elastic/beats/v7/libbeat/cmd/instance.Run.func1.1\n\t/Users/daveops/.gvm/pkgsets/go1.18.4/global/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20220726163833-e8a0da132f1f/libbeat/cmd/instance/beat.go:172\nruntime.gopanic\n\t/Users/daveops/.gvm/gos/go1.18.4/src/runtime/panic.go:838\nruntime.goPanicIndex\n\t/Users/daveops/.gvm/gos/go1.18.4/src/runtime/panic.go:89\ngithub.com/elastic/cloudbeat/resources/fetchersManager.addCredentialsToFetcherConfiguration\n\t/Users/daveops/git_repos/elastic/cloudbeat/resources/fetchersManager/factory.go:109\ngithub.com/elastic/cloudbeat/resources/fetchersManager.(*factories).ParseConfigFetchers\n\t/Users/daveops/git_repos/elastic/cloudbeat/resources/fetchersManager/factory.go:67\ngithub.com/elastic/cloudbeat/beater.initRegistry\n\t/Users/daveops/git_repos/elastic/cloudbeat/beater/cloudbeat.go:217\ngithub.com/elastic/cloudbeat/beater.NewCloudbeat\n\t/Users/daveops/git_repos/elastic/cloudbeat/beater/cloudbeat.go:99\ngithub.com/elastic/cloudbeat/launcher.(*launcher).runBeater\n\t/Users/daveops/git_repos/elastic/cloudbeat/launcher/launcher.go:127\ngithub.com/elastic/cloudbeat/launcher.(*launcher).run\n\t/Users/daveops/git_repos/elastic/cloudbeat/launcher/launcher.go:105\ngithub.com/elastic/cloudbeat/launcher.(*launcher).Run\n\t/Users/daveops/git_repos/elastic/cloudbeat/launcher/launcher.go:99\ngithub.com/elastic/beats/v7/libbeat/cmd/instance.(*Beat).launch\n\t/Users/daveops/.gvm/pkgsets/go1.18.4/global/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20220726163833-e8a0da132f1f/libbeat/cmd/instance/beat.go:475\ngithub.com/elastic/beats/v7/libbeat/cmd/instance.Run.func1\n\t/Users/daveops/.gvm/pkgsets/go1.18.4/global/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20220726163833-e8a0da132f1f/libbeat/cmd/instance/beat.go:180\ngithub.com/elastic/beats/v7/libbeat/cmd/instance.Run\n\t/Users/daveops/.gvm/pkgsets/go1.18.4/global/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20220726163833-e8a0da132f1f/libbeat/cmd/instance/beat.go:181\ngithub.com/elastic/beats/v7/libbeat/cmd.genRunCmd.func1\n\t/Users/daveops/.gvm/pkgsets/go1.18.4/global/pkg/mod/github.com/elastic/beats/v7@v7.0.0-alpha2.0.20220726163833-e8a0da132f1f/libbeat/cmd/run.go:36\ngithub.com/spf13/cobra.(*Command).execute\n\t/Users/daveops/.gvm/pkgsets/go1.18.4/global/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:876\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/Users/daveops/.gvm/pkgsets/go1.18.4/global/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990\ngithub.com/spf13/cobra.(*Command).Execute\n\t/Users/daveops/.gvm/pkgsets/go1.18.4/global/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918\nmain.main\n\t/Users/daveops/git_repos/elastic/cloudbeat/main.go:29\nruntime.main\n\t/Users/daveops/.gvm/gos/go1.18.4/src/runtime/proc.go:250","ecs.version":"1.6.0"}`

This bug was fixed - good catch!

**Review by DaveSys911 — APPROVED** [review]
Looks great and work.Two things:

1. Please make sure to set indention in your ide to tabs otherwise the configmap generation looks weird.(see below)
image


2. I've fixed the indention issue and added a small enhancement to have Kustomize pickup the aws creds automatically from the local host env vars. I branched out from your branch u can find it here https://github.com/DaveSys911/cloudbeat/tree/kustomize-review-fixes

**Review by DaveSys911 — APPROVED** [review]
LGTM

**Pull request merged**